lib/elf/file.rb

```diff
@@ -374,12 +374,16 @@ def is_compatible(other)
       raise TypeError.new("wrong argument type #{other.class} (expected Elf::File)") unless
         other.is_a? Elf::File
 
+      compatible_abi = case
+        when [@abi, other.abi].include?(Elf::OsAbi::SysV); true
```

Comment:
That's interesting; to be honest over here I only have SYSV ELFs
everywhere, so I'm not sure why you would have different ABI generated on
your system… and I wonder whether the loader actually tests anything at all.